#!/usr/bin/env python
PACKAGE = "franka_example_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#          name                   paramtype level desciption                    default=? min  max
gen.add("AC_Md", double_t, 0, "AC_Md", 2, 0, 1000)#
gen.add("AC_Dd", double_t, 0, "AC_Dd", 27,  0, 1000)#
gen.add("AC_Kd", double_t, 0, "AC_Kd", 80,  0, 2000)#

gen.add("Ki_pos", double_t, 0, "Ki_pos", 1, 0, 100)#
gen.add("Kp_pos", double_t, 0, "Kp_pos", 1000,  0, 5000)#
gen.add("Kv_pos", double_t, 0, "Kv_pos", 30,  0, 100)#

gen.add("Ki_ori", double_t, 0, "Ki_ori", 0, 0, 10)#
gen.add("Kp_ori", double_t, 0, "Kp_ori", 500,  0, 1000)#
gen.add("Kv_ori", double_t, 0, "Kv_ori", 1,  0, 100)#

gen.add("Kp_ns", double_t, 0, "Kp_ns", 40, 0, 1000)#
gen.add("Kv_ns", double_t, 0, "Kv_ns", 1, 0, 100)#

gen.add("rd", double_t, 0, "rd", 0.000001, 0, 10)#

#pkgname nodename name
exit(gen.generate(PACKAGE, "dynamic_compliance", "admittance_controller_param"))
